lib/fluent/plugin/checkpoint_metadata_cache.rb

```diff
@@ -0,0 +1,92 @@
+#
+# Fluentd Kubernetes Metadata Filter Plugin - Enrich Fluentd events with
+# Kubernetes metadata
+#
+# Copyright 2017 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+require "sqlite3"
+
+module KubernetesMetadata
+  module CheckPointCache
+    def start_checkpoint
+      while @checkpoint_thread_running
+        log.trace "check pointing cache to disk" if log.trace?
+        prune_old_entries
+        write_cache_to_file
+        sleep @checkpoint_interval
+      end
+    end
+
+    def initialize_db
+      @db = SQLite3::Database.new @checkpoint_db_path
+      @db.execute <<-SQL
+        create table IF NOT EXISTS cache (
+          id TEXT PRIMARY KEY,
+          val TEXT,
```

Comment:
Is text sufficiently large enough to hold a well formed json blob?

Reply:
The max value is 2^32 − 1, this _can_ fail in cases with pods with large labels. I will add in a skip for overly large values.